experiment/urlgetter/configurer.go

```diff
@@ -72,6 +72,7 @@ func (c Configurer) NewConfiguration() (Configuration, error) {
 	}
 	dnsclient, err := netx.NewDNSClient(
 		configuration.HTTPConfig, c.Config.ResolverURL,
+		c.Config.HTTPHost, c.Config.TLSServerName,
```

Comment:
I suggest to implement changes leading to something similar to:

```suggestion
	dnsclient, err := netx.NewDNSClientWithOverrides(
		configuration.HTTPConfig, c.Config.ResolverURL,
		c.Config.DNSHTTPHost, c.Config.DNSTLSServerName,
```

That is, I am proposing the following:

1. `NewDNSClientWithOverrides` takes four arguments and `NewDNSClient` becomes a wrapper that calls `NewDNSClientWithOverrides` with the last two arguments set to `""`. This way, we need to change less parts of the codebase and, also, the caller of `NewDNSClient` does not need to be concerned about the last two arguments set to `""` when these arguments are not required.

2. We introduce `DNSHTTPHost` and `DNSTLSServerName` in `urlgetter.Config` so that someone could configure _different_ overrides for configuring the DNS and for configuring post-DNS activities when using URL getter. If we do not apply the latter change, setting `HTTPHost` for the sake of the DNS will also have impact on any HTTP request subsequently set by the `urlgetter`, which could break our testing in some cases. For example, we could set a specific SNI for DNS when using DoH, assuming that it _may_ break, and the `urlgetter` target is `https://example.com`. For some reason, the DoH step is successful and now we try to access `example.com` but we fail because of the SNI that was configured for the DNS.